src/main/java/com/znsio/rpi/utils/ScreenShotManager.java

```diff
@@ -0,0 +1,101 @@
+package com.znsio.rpi.utils;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.log4j.Logger;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+public class ScreenShotManager {
+    private static final String directoryPath = System.getProperty("user.dir") + File.separator +
+            "TestReport/Screenshots";
+    private static final Logger LOGGER = Logger.getLogger(ScreenShotManager.class.getSimpleName());
+
+    public static String captureScreenShot(WebDriver webDriver) {
+        String testName = Thread.currentThread().getStackTrace()[3].getMethodName();
```

Comment:
Except the logMessage, everything seems to be duplicated. Can we move the duplication?

Reply:
I actually though about this, and there is one problem: The stack trace to getMethodName will become different if I'm calling one method from another and it won't give intended method name

Reply:
Addressed this. Also implemented the captureAndAttachScreenshot method in ReportPortalLogger so that consumer doesn't have to import ScreenShotManager for logging screenshots to ReportPortal